edb/schema/referencing.py

```diff
@@ -164,15 +182,16 @@ def derive_ref(
             parent_cmd.add(cmd)
             schema = delta.apply(schema, context)
 
-        derived = schema.get(derived_name)
+        derived: ReferencedT = schema.get(derived_name)
 
         return schema, derived
 
 
 class ReferencedInheritingObject(inheriting.InheritingObject,
                                  ReferencedObject):
-
-    def get_implicit_bases(self, schema):
+    def get_implicit_bases(self,
```

Comment:
The return type would be the same as `self`, so making this generic (ie. `self: ReferencedInheritingObjectT`) would improve the ergonomics of usage of this method.